File: src/main/java/back/ecommerce/dto/request/product/ProductSearchConditionRequest.java
package back.ecommerce.dto.request.product;

import java.util.Map;

import back.ecommerce.domain.product.Category;
import lombok.Getter;

@Getter
public class ProductSearchConditionRequest {
	private final Category category;
	private final Map<String, String> parameters;

	private ProductSearchConditionRequest(Category category, Map<String, String> parameters) {
		this.category = category;
		this.parameters = parameters;
	}

	public static ProductSearchConditionRequest create(Category category, Map<String, String> parameters) {
		validateEmptyCondition(parameters);
		return new ProductSearchConditionRequest(category, parameters);
	}

	private static void validateEmptyCondition(Map<String, String> parameters) {
		parameters.putIfAbsent("name", null);
		parameters.putIfAbsent("brandName", null);
		parameters.putIfAbsent("minPrice", null);
		parameters.putIfAbsent("maxPrice", null);
		parameters.putIfAbsent("sort", "new");
		parameters.putIfAbsent("page", "");
	}
}

File: src/main/java/back/ecommerce/product/dto/request/ProductSearchConditionRequest.kt
package back.ecommerce.product.dto.request

import back.ecommerce.product.entity.Category

class ProductSearchConditionRequest private constructor(val category: Category, val parameters: Map<String, String?>) {
    companion object {
        fun create(category: Category, parameters: MutableMap<String, String?>): ProductSearchConditionRequest {
            validateEmptyCondition(parameters)
            return ProductSearchConditionRequest(category, parameters)
        }

        private fun validateEmptyCondition(parameters: MutableMap<String, String?>) {
            parameters.putIfAbsent("name", null)
            parameters.putIfAbsent("brandName", null)
            parameters.putIfAbsent("minPrice", null)
            parameters.putIfAbsent("maxPrice", null)
            parameters.putIfAbsent("sort", "new")
            parameters.putIfAbsent("page", "")
        }
    }
}
